client/components/Cart.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Component } from 'react';
+import { fetchOrders } from '../store/products'
+
+
+// Displays single product
+class Cart extends Component {
+    async componentDidMount () {
+        // Fetch product data
+        console.log('THE PROPS', this.props);
+        const orderItems = await this.props.orderItems(this.props.auth.id);        
+        //console.log(userOrders);
+        
+     //   this.props.init(this.props.match.params.id);
+    }
+
+    render () {
+        if(!this.props.products.userOrders){
+            return (
+                <p>...loading</p>
+            )
```

Comment:
on first render our array is empty, so this will prevent mapping over an empty array

Reply:
We don't need this. Since we are mapping, even if the array is empty, it will just never return anything. Then the componentDidMount will run, at which point it will map over the now-filled array. Side note though, I am going to work on a loading component and potentially put in really small delays on our thunks so that the site appears to be loading data.